easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -449,6 +449,13 @@ def build_step(self):
                 env.setvar("CMAKE_INCLUDE_PATH", include_paths)
                 env.setvar("CMAKE_LIBRARY_PATH", library_paths)
 
+            curr_cc = os.getenv('CC')
+            python_ldshared = get_config_vars('LDSHARED')[0].split(' ')
+            if python_ldshared[0] != curr_cc:
```

Comment:
@Micket Missing import statement for `get_config_vars`?

Also, I would like to be a bit more careful here. We're assuming that there's at least one entry in the return value of `get_config_vars`. Will that always be the case, on all platforms?

To avoid introducing problems when installing any Python package, I would prefer a bit more robust code, like:

```python
curr_cc = os.getenv('CC')
python_ldshared = get_config_vars('LDSHARED')
if python_ldshared:
    if python_ldshared[0].split(' ')[0] != curr_cc:
        ...
else:
    self.log.warning("Failed to determine value for Python's $LDSHARED, so can't check it against $CC ('%s')", curr_cc)
```

Reply:
Or set it to `$CC -shared` in the else condition?
I have to note that distutils will append both LDFLAGS and CFLAGS to this command.

I.M.H.O. distutils is a bit of a hodgepodge of conditional code so it's hard to say what's correct. 
As far as I can see, it always expects get_config_vars to return something, else it would append LDFLAGS to an empty string?

If there truly is a case where it's not set at all, what will it use to link? Surely `$CC -shared` would be as good as any choice then.

Reply:
@Micket That sounds reasonable, let's go with that.